### PR TITLE
Add Congestion Control Considerations

### PR DESCRIPTION
**Pull request by markusa**

### CODE REVIEW

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 1]
```diff
@@ -1034,6 +1054,23 @@ If a subflow attempts to join an existing MP-DCCP connection, but MP-DCCP option
...
+Senders MUST manage per-path congestion status, and SHOULD avoid to send
+more data on a given path than congestion control on that path
+allows.
+
+When a Multipath DCCP connection uses two or more paths, there is no
+guarantee that these paths are fully disjoint.  When two (or more
+paths) share the same bottleneck, using a standard congestion control
+scheme could result in an unfair distribution of the bandwidth with
+the multipath connection getting more bandwidth than competing single
+paths connections.  Multipath TCP uses the LIA congestion control
```
Comment (dirkhugo) [inline draft-ietf-tsvwg-multipath-dccp.mkd:1068]: ... uses the coupled congestion control scheme 'Linked Increases Algorithm' specified in ...

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 2]
```diff
@@ -1034,6 +1054,23 @@ If a subflow attempts to join an existing MP-DCCP connection, but MP-DCCP option
...
+scheme specified in [RFC6356] to solve this problem.  This scheme can
+immediately be adapted to Multipath DCCP.  The same applies to other
+coupled congestion control schemes, which have been proposed for
+Multipath TCP such as [OLIA].
+
```
Comment (dirkhugo) [inline draft-ietf-tsvwg-multipath-dccp.mkd:1073]: ... such as Opportunistic Linked Increases Algorithm (OLIA) [OLIA].

**Review by dirkhugo — APPROVED** [review]
I would add an explanation of the abbreviations as commented and maybe explain why they are applicable also to MP-DCCP, since at least LIA "algorithm relies on the traditional TCP mechanisms to detect drops, to retransmit data, etc." - and DCCP does not provide automatic retransmission (not sure whether it matters)

File: draft-ietf-tsvwg-multipath-dccp.mkd (hunk 1 above)
Comment (markusa) [reply draft-ietf-tsvwg-multipath-dccp.mkd]: https://github.com/markusa/ietf-multipath-dccp/pull/54/commits/2f016c6b698dd809ea2d8d81a5b3129b3eb6cb29

File: draft-ietf-tsvwg-multipath-dccp.mkd (hunk 2 above)
Comment (markusa) [reply draft-ietf-tsvwg-multipath-dccp.mkd:1074]: https://github.com/markusa/ietf-multipath-dccp/pull/54/commits/2f016c6b698dd809ea2d8d81a5b3129b3eb6cb29

**Comment by markusa** [comment]
> Maybe explain why they are applicable also to MP-DCCP, since at least LIA "algorithm relies on the traditional TCP mechanisms to detect drops, to retransmit data, etc." - and DCCP does not provide automatic retransmission (not sure whether it matters)

Re-transmission doesn't matter. Important is the acknowledgement of data, which is the same for TCP and DCCP.

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 3]
```diff
@@ -1034,6 +1054,24 @@ If a subflow attempts to join an existing MP-DCCP connection, but MP-DCCP option
...
+When a Multipath DCCP connection uses two or more paths, there is no
+guarantee that these paths are fully disjoint.  When two (or more
+paths) share the same bottleneck, using a standard congestion control
+scheme could result in an unfair distribution of the bandwidth with
+the multipath connection getting more bandwidth than competing single
+paths connections.  Multipath TCP uses the coupled congestion control
+scheme Linked Increases Algorithm (LIA) congestion control
+scheme specified in [RFC6356] to solve this problem.  This scheme can
+immediately be adapted to Multipath DCCP.  The same applies to other
```
[the remark this answers is not in the document]
Comment (markusa) [reply draft-ietf-tsvwg-multipath-dccp.mkd:1071]: https://github.com/markusa/ietf-multipath-dccp/commit/26644b4ddbb0a5fc70a175e40e15b5c266b9dac4

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 4]
```diff
@@ -1034,6 +1054,24 @@ If a subflow attempts to join an existing MP-DCCP connection, but MP-DCCP option
...
+more data on a given path than congestion control on that path
+allows.
+
+When a Multipath DCCP connection uses two or more paths, there is no
+guarantee that these paths are fully disjoint.  When two (or more
+paths) share the same bottleneck, using a standard congestion control
+scheme could result in an unfair distribution of the bandwidth with
+the multipath connection getting more bandwidth than competing single
+paths connections.  Multipath TCP uses the coupled congestion control
+scheme Linked Increases Algorithm (LIA) congestion control
```
[the remark this answers is not in the document]
Comment (markusa) [reply draft-ietf-tsvwg-multipath-dccp.mkd:1069]: https://github.com/markusa/ietf-multipath-dccp/commit/26644b4ddbb0a5fc70a175e40e15b5c266b9dac4

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 5]
```diff
@@ -1034,6 +1054,24 @@ If a subflow attempts to join an existing MP-DCCP connection, but MP-DCCP option
...
+Senders MUST manage per-path congestion status, and SHOULD avoid to send
+more data on a given path than congestion control on that path
+allows.
+
+When a Multipath DCCP connection uses two or more paths, there is no
+guarantee that these paths are fully disjoint.  When two (or more
+paths) share the same bottleneck, using a standard congestion control
+scheme could result in an unfair distribution of the bandwidth with
+the multipath connection getting more bandwidth than competing single
+paths connections.  Multipath TCP uses the coupled congestion control
```
[the remark this answers is not in the document]
Comment (markusa) [reply draft-ietf-tsvwg-multipath-dccp.mkd:1068]: https://github.com/markusa/ietf-multipath-dccp/commit/26644b4ddbb0a5fc70a175e40e15b5c266b9dac4

[the remark this answers is not in the document]
Comment (dirkhugo) [reply draft-ietf-tsvwg-multipath-dccp.mkd:1068]: fine for me - just would put the bracket before paths: ... when two (or more) paths share ...
and below use singular: ... competing single path connections. ...

**Pull request merged**